library/src/main/java/com/datatorrent/lib/parser/XmlParser.java

```diff
@@ -0,0 +1,139 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.datatorrent.lib.parser;
+
+import java.io.IOException;
+import java.io.StringReader;
+
+import javax.xml.XMLConstants;
+import javax.xml.bind.JAXBContext;
+import javax.xml.bind.JAXBElement;
+import javax.xml.bind.JAXBException;
+import javax.xml.bind.Unmarshaller;
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.transform.stream.StreamSource;
+import javax.xml.validation.Schema;
+import javax.xml.validation.SchemaFactory;
+import javax.xml.validation.Validator;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.w3c.dom.Document;
+import org.xml.sax.SAXException;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FSDataInputStream;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+
+
+import com.datatorrent.api.DefaultOutputPort;
+import com.datatorrent.netlet.util.DTThrowable;
+
+public class XmlParser extends Parser<String>
+{
+  private String schemaXSDFile;
+  private transient Unmarshaller unmarshaller;
+  private transient Validator validator;
+
+  public transient DefaultOutputPort<Document> parsedOutput = new DefaultOutputPort<Document>();
+
+  @Override
+  public Object convert(String tuple)
+  {
+    // This method is not invoked for XML parser
+    return null;
+  }
+
+  @Override
+  public void processTuple(String inputTuple)
+  {
+    try {
+      if (out.isConnected()) {
+        StringReader reader = new StringReader(inputTuple);
+        JAXBElement<?> output = unmarshaller.unmarshal(new StreamSource(reader), getClazz());
+        LOG.debug(output.getValue().toString());
+        emittedObjectCount++;
+        out.emit(output.getValue());
+      } else {
+        validator.validate(new StreamSource(inputTuple));
```

Comment:
Unmarshal does validation as well, so no need to explicitly validate in POJO case.
Yes, validate should only be applied if Schema file is specified. Will fix that.